src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs

```diff
@@ -22,19 +22,21 @@ public static class ResolverUtility
             ConcurrentDictionary<LibraryRangeCacheKey, Task<GraphItem<RemoteResolveResult>>> cache,
             LibraryRange libraryRange,
             NuGetFramework framework,
+            string runtimeIdentifier,
             GraphEdge<RemoteResolveResult> outerEdge,
             RemoteWalkContext context,
             CancellationToken cancellationToken)
         {
             var key = new LibraryRangeCacheKey(libraryRange, framework);
 
             return cache.GetOrAdd(key, (cacheKey) =>
-                FindLibraryEntryAsync(cacheKey.LibraryRange, framework, outerEdge, context, cancellationToken));
+                FindLibraryEntryAsync(cacheKey.LibraryRange, framework, runtimeIdentifier, outerEdge, context, cancellationToken));
         }
 
         public static async Task<GraphItem<RemoteResolveResult>> FindLibraryEntryAsync(
             LibraryRange libraryRange,
             NuGetFramework framework,
+            string runtimeIdentifier,
```

Comment:
I don't think you need the runtime identifier per the other comment.

Reply:
We do because we're being smart enough to not just copy all target graphs across tfm and tfm+rid in new `packages.lock.json` file rather tfm+rid will only have the changed dependencies in tfm+rid.

So when we come here to resolve all dependencies for tfm+rid, we need to understand the rid in order to get the corresponding dependency. 

Go through the code changes of `ResolverUtility` if you still dont understand then let me know i'll try to add more comments there to make it understandable.

Reply:
I was under the impression that we don't need them and you never truly use them. 

I'll bug you offline if I reach the same conclusion when I go over it again.

Reply:
I still think this can be changed. 
Let's discuss offline.